backend/app/services/knowledge/structured/sql_validator.py

```
@@ -0,0 +1,217 @@
+
+        # Check 2: Forbidden keywords
+        for keyword in self.FORBIDDEN_KEYWORDS:
+            # Use word boundary to avoid false positives (e.g., "UPDATED" vs "UPDATE")
+            pattern = rf"\b{keyword}\b"
+            if re.search(pattern, sql_upper):
+                errors.append(f"Forbidden operation: {keyword}")
+
+        # Check 3: SQL injection patterns
+        for pattern in self.INJECTION_PATTERNS:
+            if re.search(pattern, sql_upper):
+                errors.append(f"Potential SQL injection detected")
```

Ruff (0.15.0)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)